rpc/server.go

```diff
@@ -82,72 +83,52 @@ func (rs *RpcServer) registerHandlers() (err error) {
 
 		switch serde.RequestMethod(validationResult.Method) {
 		case serde.GetAddressMethod:
-			return processRequest(rs, requestData, func(T serde.NoPayloadRequest) string {
-				return rs.client.Address.Hex()
+			return processRequest(rs, requestData, func(T serde.NoPayloadRequest) (string, error) {
+				return rs.client.Address.Hex(), nil
 			})
 		case serde.VersionMethod:
-			return processRequest(rs, requestData, func(T serde.NoPayloadRequest) string {
-				return rs.client.Version()
+			return processRequest(rs, requestData, func(T serde.NoPayloadRequest) (string, error) {
+				return rs.client.Version(), nil
 			})
 		case serde.DirectFundRequestMethod:
-			return processRequest(rs, requestData, func(obj directfund.ObjectiveRequest) directfund.ObjectiveResponse {
-				return rs.client.CreateLedgerChannel(obj.CounterParty, obj.ChallengeDuration, obj.Outcome)
+			return processRequest(rs, requestData, func(obj directfund.ObjectiveRequest) (directfund.ObjectiveResponse, error) {
+				return rs.client.CreateLedgerChannel(obj.CounterParty, obj.ChallengeDuration, obj.Outcome), nil
 			})
 		case serde.DirectDefundRequestMethod:
-			return processRequest(rs, requestData, func(obj directdefund.ObjectiveRequest) protocols.ObjectiveId {
-				return rs.client.CloseLedgerChannel(obj.ChannelId)
+			return processRequest(rs, requestData, func(obj directdefund.ObjectiveRequest) (protocols.ObjectiveId, error) {
+				return rs.client.CloseLedgerChannel(obj.ChannelId), nil
 			})
 		case serde.VirtualFundRequestMethod:
-			return processRequest(rs, requestData, func(obj virtualfund.ObjectiveRequest) virtualfund.ObjectiveResponse {
-				return rs.client.CreateVirtualPaymentChannel(obj.Intermediaries, obj.CounterParty, obj.ChallengeDuration, obj.Outcome)
+			return processRequest(rs, requestData, func(obj virtualfund.ObjectiveRequest) (virtualfund.ObjectiveResponse, error) {
+				return rs.client.CreateVirtualPaymentChannel(obj.Intermediaries, obj.CounterParty, obj.ChallengeDuration, obj.Outcome), nil
 			})
 		case serde.VirtualDefundRequestMethod:
-			return processRequest(rs, requestData, func(obj virtualdefund.ObjectiveRequest) protocols.ObjectiveId {
-				return rs.client.CloseVirtualChannel(obj.ChannelId)
+			return processRequest(rs, requestData, func(obj virtualdefund.ObjectiveRequest) (protocols.ObjectiveId, error) {
+				return rs.client.CloseVirtualChannel(obj.ChannelId), nil
 			})
 		case serde.PayRequestMethod:
-			return processRequest(rs, requestData, func(obj serde.PaymentRequest) serde.PaymentRequest {
+			return processRequest(rs, requestData, func(obj serde.PaymentRequest) (serde.PaymentRequest, error) {
 				rs.client.Pay(obj.Channel, big.NewInt(int64(obj.Amount)))
-				return obj
+				return obj, nil
```

Comment:
I may be missing something, but it looks like a lot of these returned errors are set to `nil`? 

In some cases, that seems to make sense (returning the version for example). In other cases, it seems that we need to return an error (making a payment for example, in the case of insufficient funds). Is there more work to do to get those errors generated / wired up?

Reply:
Currently the underlying `client` methods do not all return an error object. For the ones that don't, I just assigned a `nil` error for now to satisfy the newly updated `processPayload` return types. I believe the underlying `client` methods should be updated to all return an error object. If you agree, and it won't create a breaking change for using go-nitro as a library, I can either include that update as part of this PR or a follow-up one

Reply:
Thanks, that makes sense! 

A follow-up PR would be great 🙏